Rust 1.89.0 removes support for lint level attributes for `text_direction_codepoint_in_literal` except at the crate level.
https://github.com/rust-lang/rust/pull/141004
https://aur.archlinux.org/packages/anki#comment-1035765
From: Robert H <robert@jadedpasta.net>
--- a/rslib/i18n/src/generated.rs
+++ b/rslib/i18n/src/generated.rs
@@ -4,6 +4,5 @@
 // Include auto-generated content
 
 #![allow(clippy::all)]
-#![allow(text_direction_codepoint_in_literal)]
 
 include!(concat!(env!("OUT_DIR"), "/strings.rs"));
--- a/rslib/i18n/src/lib.rs
+++ b/rslib/i18n/src/lib.rs
@@ -1,6 +1,8 @@
 // Copyright: Ankitects Pty Ltd and contributors
 // License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
 
+#![allow(text_direction_codepoint_in_literal)]
+
 mod generated;
 
 use std::borrow::Cow;
